`:has-attachment` pseudo-class should be prefixed

nt1m opened the pull request:
#### ee0b8ecdb38dbf2891eaf5844f5214540d43f703
<pre>
`:has-attachment` pseudo-class should be prefixed
<a href="https://bugs.webkit.org/show_bug.cgi?id=266948">https://bugs.webkit.org/show_bug.cgi?id=266948</a>
<a href="https://rdar.apple.com/120599170">rdar://120599170</a>

Reviewed by Darin Adler.

This pseudo-class is only used by the Apple Mail apps along with the &lt;attachment&gt; element feature.

Prefix it with `-apple-` to make this clear and subject to validation for `-apple-` prefixes.

* LayoutTests/fast/attachment/has-attachment-pseudo-class-expected.txt: Renamed from LayoutTests/fast/css/has-attachment-expected.txt.
* LayoutTests/fast/attachment/has-attachment-pseudo-class.html: Renamed from LayoutTests/fast/css/has-attachment.html.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/wincairo/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:

Move test to the fast/attachment directory which is already skipped on relevant platforms.

* Source/WebCore/css/CSSPseudoSelectors.json:
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::checkOne const):
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::addPseudoClassType):

Canonical link: <a href="https://commits.webkit.org/272820@main">https://commits.webkit.org/272820@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c23836f574646ca47bfcb1468f51b6495e304911

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32895 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11653 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34772 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35519 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29720 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33841 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13993 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8829 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29163 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33351 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9817 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29376 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8555 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8699 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29332 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36851 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29875 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29741 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34834 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8830 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6783 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32689 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10521 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7690 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9435 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9454 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
I understand why this isn't needed on the web and should not be web exposed but not why this is Apple specific. Seems like others might want to make apps that work like this.

nt1m committed:
`:has-attachment` pseudo-class should be prefixed
https://bugs.webkit.org/show_bug.cgi?id=266948
rdar://120599170

Reviewed by Darin Adler.

This pseudo-class is only used by the Apple Mail apps along with the <attachment> element feature.

Prefix it with `-apple-` to make this clear and subject to validation for `-apple-` prefixes.

* LayoutTests/fast/attachment/has-attachment-pseudo-class-expected.txt: Renamed from LayoutTests/fast/css/has-attachment-expected.txt.
* LayoutTests/fast/attachment/has-attachment-pseudo-class.html: Renamed from LayoutTests/fast/css/has-attachment.html.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/wincairo/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:

Move test to the fast/attachment directory which is already skipped on relevant platforms.

* Source/WebCore/css/CSSPseudoSelectors.json:
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::checkOne const):
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::addPseudoClassType):

Canonical link: https://commits.webkit.org/272820@main
